src/System.Security.Cryptography.OpenSsl/src/System/Security/Cryptography/RSAOpenSsl.cs

```diff
@@ -45,6 +45,31 @@ public RSAOpenSsl(RSAParameters parameters)
             ImportParameters(parameters);
         }
 
+        /// <summary>
+        /// Create an RSAOpenSsl from an existing <see cref="IntPtr"/> whose value is an
+        /// existing OpenSSL <c>RSA*</c>.
+        /// </summary>
+        /// <remarks>
+        /// This method will increase the reference count of the <c>RSA*</c>, the caller should
+        /// continue to manage the lifetime of their reference.
+        /// </remarks>
+        /// <param name="handle">A pointer to an OpenSSL <c>RSA*</c></param>
+        /// <exception cref="ArgumentException"><paramref name="handle" /> is invalid</exception>
+        public RSAOpenSsl(IntPtr handle)
+        {
+            if (handle == IntPtr.Zero)
+                throw new ArgumentException(SR.Cryptography_OpenInvalidHandle, "handle");
+
+            _legalKeySizesValue = new[] { s_legalKeySizes };
```

Comment:
It's actually a protected field from [RSA](https://github.com/dotnet/corefx/blob/588aea39e878f696f3e93af6a167855c5397c5f6/src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/RSA.cs#L22) that the derived types need to set before exiting the constructor.

(Or, I suppose, by overriding the LegalKeySizes accessor and just cloning my existing static readonly.)